src/Calendar/ICalProvider.php

```diff
@@ -53,35 +53,39 @@ public function icalSingle(Activity $activity): string
     private function createCalendar(): Calendar
     {
         $calendar = new Calendar();
-        $calendar
-            ->setProductIdentifier('-//Helpless Kiwi//'.($_ENV['ORG_NAME'] ?? 'kiwi').' v1.0//NL')
-            ->addTimeZone(new TimeZone(date_default_timezone_get()))
-        ;
-
-        return $calendar;
+        return $calendar
+            ->setProductIdentifier('-//Helpless Kiwi//' . ($_ENV['ORG_NAME'] ?? 'kiwi') . ' v1.0//NL')
+            ->addTimeZone(new TimeZone(date_default_timezone_get()));
     }
 
     private function createEvent(Activity $activity): Event
     {
+        assert($activity->getLocation() !== null);
```

Comment:
These asserts call a function, but with a function call, it is not guaranteed that later calls will return the same value. Therefore, I'd propose the following modification:
```php
$address = $activity->getLocation()?->getAddress();
assert($address !== null);
$location = new Location($address);
```

Reply:
Note that this comment is applicable throughout this PR, as I see more instances where a non-null `assert` is performed on a function.